libnetwork/internal/nftables/nftables_linux.go

```diff
@@ -0,0 +1,737 @@
+// manipulated via exported methods.
+type set struct {
+	table           *table
+	Name            string
+	ElementType     nftType
+	Flags           []string
+	Elements        map[string]struct{}
+	Dirty           bool // New set, needs to be flushed (not set when elements are added/deleted).
+	AddedElements   map[string]struct{}
+	DeletedElements map[string]struct{}
+}
+
+// SetRef is a handle for an nftables named set.
+type SetRef struct {
+	s *set
+}
+
+// PrefixSet creates a new named nftables set for IPv4 or IPv6 addresses (depending
+// on the address family of the [TableRef]), and returns its [SetRef]. Or, if the
+// set has already been created, its [SetRef] is returned.
+//
+// ([TableRef] does not support "inet", only "ip" or "ip6". So the element type can
+// always be determined. But, there's no "inet" element type, so this will need to
+// change if we need an "inet" table.)
+//
+// See https://wiki.nftables.org/wiki-nftables/index.php/Sets#Named_sets
+func (t TableRef) PrefixSet(name string) SetRef {
+	if s, ok := t.t.Sets[name]; ok {
+		return SetRef{s}
+	}
+	s := &set{
+		table:           t.t,
+		Name:            name,
+		Elements:        map[string]struct{}{},
+		ElementType:     nftTypeIPv4Addr,
+		Flags:           []string{"interval"},
+		Dirty:           true,
+		AddedElements:   map[string]struct{}{},
+		DeletedElements: map[string]struct{}{},
+	}
+	if t.t.Family == IPv6 {
+		s.ElementType = nftTypeIPv6Addr
+	}
+	t.t.Sets[name] = s
+	log.G(context.TODO()).WithFields(log.Fields{
+		"family": t.t.Family,
+		"table":  t.t.Name,
+		"set":    name,
+	}).Debug("nftables: created set")
+	return SetRef{s}
+}
+
+// AddElement adds an element to a set. It is the caller's responsibility to make sure
+// the element has the correct type. It is an error to add an element that is already
+// in the set.
+func (s SetRef) AddElement(element string) error {
+	if _, ok := s.s.Elements[element]; ok {
+		return fmt.Errorf("set already contains element %q", element)
+	}
+	s.s.Elements[element] = struct{}{}
+	s.s.AddedElements[element] = struct{}{}
+	log.G(context.TODO()).WithFields(log.Fields{
+		"family":  s.s.table.Family,
+		"table":   s.s.table.Name,
+		"set":     s.s.Name,
+		"element": element,
+	}).Debug("nftables: added set element")
+	return nil
+}
+
+// DeleteElement deletes an element from the set. It is an error to delete an
+// element that is not in the set.
+func (s SetRef) DeleteElement(element string) error {
+	if _, ok := s.s.Elements[element]; !ok {
+		return fmt.Errorf("set does not contain element %q", element)
+	}
+	delete(s.s.Elements, element)
+	s.s.DeletedElements[element] = struct{}{}
+	log.G(context.TODO()).WithFields(log.Fields{
+		"family":  s.s.table.Family,
+		"table":   s.s.table.Name,
+		"set":     s.s.Name,
+		"element": element,
+	}).Debug("nftables: deleted set element")
+	return nil
+}
+
+//////////////////////////////
+// Internal
+
+/* Can't make text/template range over this, not sure why ...
```

Comment:
Is this something we need to keep, or can it be deleted?

Reply:
An alternative could be to remove it, and open a draft PR with the code added back (for the "reminder").